common/src/main/java/com/microsoft/identity/common/internal/net/ObjectMapper.java

```diff
@@ -47,6 +58,124 @@
      */
     public static final String ENCODING_SCHEME = "UTF-8";
     public static final String TAG = ObjectMapper.class.getSimpleName();
+    public static final Gson GSON = new GsonBuilder()
+            .registerTypeAdapterFactory(new UnknownParamTypeAdapterFactory())
+            .create();
+
+    public static class UnknownParamTypeAdapterFactory implements TypeAdapterFactory {
```

Comment:
```suggestion
    /*
     *This likely deserves a comment.  What I'm doing here is hooking the underlying GSON implementation's
     * use of JsonReader.  I KNOW that what's happening in the type adapter that we're delegating to is that if the
     * name of the field isn't in the names that they mapped, they skip it.  What we're doing instead is to keep
     * track of the last seen name, and then when we get called to skip a value, if it will be a string, save it in a map.
     * This map is linked to preserve the order of the parameters for testing use.  At the end of this process, we
     * store the resulting Iterable in the object that can accept it.
     */
    public static class UnknownParamTypeAdapterFactory implements TypeAdapterFactory {
```